packages/spacecat-shared-rum-api-client/src/functions/formVitals.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { FlatBundle } from '../common/flat-bundle.js';
+
+function collectFormVitals(groupedByUrlId) {
+  const { url, items: itemsByUrlId } = groupedByUrlId;
+
+  const result = itemsByUrlId
+    .flatMap((item) => item.items)
+    .reduce((acc, cur) => {
+      // Check for 'viewblock' checkpoint
+      if (cur && cur.checkpoint === 'viewblock' && cur.source === '.form') {
+        acc.isFormViewPresent = true;
+      } else if (cur && cur.checkpoint === 'formsubmit') {
+        acc.isFormSubmitPresent = true;
+        // for business adobe com  cur.source.toLowerCase().includes('#mktoButton_new')
+      } else if (cur && cur.checkpoint === 'click' && cur.source && /\bform\b/.test(cur.source.toLowerCase())) {
+        acc.isFormSubmitButtonClickPresent = true;
```

Comment:
how does click checkpoint containing "form" denote the presence of form submit button?

Reply:
@nit23uec -This to cover cases where submit button may be missing or some other button is used as integration

Reply:
this is not correct. the property is isFormSubmitButtonClickPresent and we must make sure that it is the submit button that we are tracking.
Else we can rename the property as `IsFormFieldClickPresent`

Reply:
Also, while checking for `isFormFieldClickPresent`, we should exclude strings having: `transform`, `perform` as we found it in few cases.

Reply:
done